cunumeric/array.py

```diff
@@ -1961,6 +2106,7 @@ def take(
                 res = res.copy()
             return res
 
```

Comment:
should we add `writable` for `out` here?

Reply:
add_boilerplate checks that. I added a change in `add_boilerplate`